openbb_platform/providers/fmp/openbb_fmp/models/economic_calendar.py

```diff
@@ -89,11 +94,46 @@ async def aextract_data(
 
         base_url = "https://financialmodelingprep.com/api/v3/economic_calendar?"
 
-        url = f"{base_url}from={query.start_date}&to={query.end_date}&apikey={api_key}"
-
-        return await amake_request(url, **kwargs)
+        # FMP allows only 3-month windows to be queried, we need to chunk to request.
+        if not query.start_date or query.start_date is None:
```

Comment:
I think that the first part of the conditional statement is all you need. The second one is redundant.

Reply:
Moved this section to `transform_query`.